exporter/loadbalancingexporter/resolver_k8s.go

```diff
@@ -0,0 +1,225 @@
+		} else {
+			logger.Error("namespace introspection error, fall back to `default` namespace", zap.Error(err))
+		}
+	}
+
+	epsSelector := fmt.Sprintf("metadata.name=%s", name)
+	epsListWatcher := &cache.ListWatch{
+		ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+			options.FieldSelector = epsSelector
+			options.TimeoutSeconds = pointer.Int64(1)
+			return clt.CoreV1().Endpoints(namespace).List(context.Background(), options)
+		},
+		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+			options.FieldSelector = epsSelector
+			options.TimeoutSeconds = pointer.Int64(1)
+			return clt.CoreV1().Endpoints(namespace).Watch(context.Background(), options)
+		},
+	}
+
+	epsStore := &sync.Map{}
+	h := &handler{endpoints: epsStore, logger: logger}
+	r := &k8sResolver{
+		logger:         logger,
+		svcName:        name,
+		svcNs:          namespace,
+		port:           ports,
+		once:           &sync.Once{},
+		endpointsStore: epsStore,
+		epsListWatcher: epsListWatcher,
+		handler:        h,
+		stopCh:         make(chan struct{}),
+	}
+	h.callback = r.resolve
+
+	return r, nil
+}
+
+func (r *k8sResolver) start(ctx context.Context) error {
+	if r.once == nil {
+		return fmt.Errorf("cannot Start() partial k8sResolver (nil *sync.Once)")
+	}
+	if r.handler == nil {
+		return fmt.Errorf("cannot Start() partial k8sResolver (nil *handler)")
+	}
+	var initErr error
+	r.once.Do(func() {
+		if r.epsListWatcher != nil {
+			r.logger.Debug("creating and starting endpoints informer")
+			epsInformer := cache.NewSharedInformer(r.epsListWatcher, &corev1.Endpoints{}, 0)
+			if _, err := epsInformer.AddEventHandler(r.handler); err != nil {
+				r.logger.Error("error adding event handler to endpoints informer", zap.Error(err))
+			}
+			go epsInformer.Run(r.stopCh)
+			if !cache.WaitForCacheSync(r.stopCh, epsInformer.HasSynced) {
+				r.logger.Error("error endpoints informer not sync")
+				initErr = errors.New("endpoints informer not sync")
+				return
+			}
+		}
+	})
+	if initErr != nil {
+		return initErr
+	}
+
+	r.logger.Debug("K8s service resolver started",
+		zap.String("service", r.svcName),
+		zap.String("namespace", r.svcNs),
+		zap.Int32s("ports", r.port))
+	return nil
+}
+
+func (r *k8sResolver) shutdown(ctx context.Context) error {
+	r.changeCallbackLock.Lock()
+	r.onChangeCallbacks = nil
+	r.changeCallbackLock.Unlock()
+
+	close(r.stopCh)
+	r.shutdownWg.Wait()
+	return nil
+}
+func newInClusterClient() (kubernetes.Interface, error) {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+	return kubernetes.NewForConfig(cfg)
+}
+
+func (r *k8sResolver) resolve(ctx context.Context) ([]string, error) {
+	r.shutdownWg.Add(1)
+	defer r.shutdownWg.Done()
+
+	var backends []string
+	r.endpointsStore.Range(func(address, value any) bool {
+		addr := address.(string)
+		if len(r.port) == 0 {
+			backends = append(backends, addr)
```

Comment:
You are not adding the port here, are you relying on the OTLP HTTP/gRPC exporter to set the default port?

Reply:
Yes, refer to the policy when dns does not provide port.